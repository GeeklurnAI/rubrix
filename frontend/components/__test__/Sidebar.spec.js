
import { shallowMount } from "@vue/test-utils";
import Sidebar from "../commons/sidebar/Sidebar";


const factory = () => {
  return shallowMount(Sidebar, {
  });
};

describe("Sidebar", () => {
  // test("mounts properly", () => {
  //   const wrapper = factory();
  //   expect(wrapper.isVueInstance()).toBeTruthy();
  // });

  test("renders properly", () => {
    const wrapper = shallowMount(Sidebar, {
      mocks: {
        $auth: {
          loggedIn: true,
        },
        $config: {
          securityEnabled: true,
        }
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});