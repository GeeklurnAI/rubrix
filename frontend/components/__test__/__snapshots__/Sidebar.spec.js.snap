// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders properly 1`] = `
<div class="sidebar">
  <div class="sidebar__actions"><a href="#" data-title="close session">
      <svgicon-stub name="logout" width="" height="" fill="true"></svgicon-stub>
    </a> <a href="#" data-title="refresh">
      <svgicon-stub name="refresh" width="" height="" fill="true"></svgicon-stub>
    </a>
    <!---->
  </div>
  <!---->
</div>
`;
